test/interpreter/date-and-time-arithmetic.spec.ts

```diff
@@ -0,0 +1,285 @@
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(true)
+    expect(engine.getCellValue(adr('D1'))).toBe(false)
+    expect(engine.getCellValue(adr('E1'))).toBe(true)
+    expect(engine.getCellValue(adr('F1'))).toBe(false)
+  })
+
+  it('compare date with datestring, different dates', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['="02/02/2020"', '02/06/2019', '=A1>B1', '=A1<B1', '=A1>=B1', '=A1<=B1', '=A1=B1', '=A1<>B1'],
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(true)
+    expect(engine.getCellValue(adr('D1'))).toBe(false)
+    expect(engine.getCellValue(adr('E1'))).toBe(true)
+    expect(engine.getCellValue(adr('F1'))).toBe(false)
+    expect(engine.getCellValue(adr('G1'))).toBe(false)
+    expect(engine.getCellValue(adr('H1'))).toBe(true)
+  })
+
+  it('compare date with datestring, the same dates', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['="02/02/2020"', '02/02/2020', '=A1>B1', '=A1<B1', '=A1>=B1', '=A1<=B1', '=A1=B1', '=A1<>B1'],
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(false)
+    expect(engine.getCellValue(adr('D1'))).toBe(false)
+    expect(engine.getCellValue(adr('E1'))).toBe(true)
+    expect(engine.getCellValue(adr('F1'))).toBe(true)
+    expect(engine.getCellValue(adr('G1'))).toBe(true)
+    expect(engine.getCellValue(adr('H1'))).toBe(false)
+  })
+
+  it('compare date with bool', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['="02/02/2020"', '=TRUE()', '=A1>B1', '=A1<B1', '=A1>=B1', '=A1<=B1'],
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(false)
+    expect(engine.getCellValue(adr('D1'))).toBe(true)
+    expect(engine.getCellValue(adr('E1'))).toBe(false)
+    expect(engine.getCellValue(adr('F1'))).toBe(true)
+  })
+
+  it('compare date with number', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['02/02/2020', '2', '=A1>B1', '=A1<B1', '=A1>=B1', '=A1<=B1'],
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(true)
+    expect(engine.getCellValue(adr('D1'))).toBe(false)
+    expect(engine.getCellValue(adr('E1'))).toBe(true)
+    expect(engine.getCellValue(adr('F1'))).toBe(false)
+  })
+
+  it('sum date with number', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['02/02/2020', '2', '=A1+B1'],
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(43865)
+  })
+
+  it('sum date with boolean', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['02/02/2020', '=TRUE()', '=A1+B1'],
+    ])
+
+    expect(engine.getCellValue(adr('C1'))).toBe(43864)
+  })
+
+  it('functions on dates', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ISEVEN("02/02/2020")', '=COS("02/02/2020")', '=BITOR("02/02/2020","16/08/1985")'],
+    ], {smartRounding: false})
+
+    expect(engine.getCellValue(adr('A1'))).toBe(false)
+    expect(engine.getCellValue(adr('B1'))).toBe(0.9965266857693633)
+    expect(engine.getCellValue(adr('C1'))).toBe(64383)
```

Comment:
Probably out of scope for this PR, I just discovered this gives different result in Excel (C1 gives `#NAME?`).

Sheet `10 ...`, file: [test.xlsx](https://github.com/handsontable/hyperformula/files/9490141/test.xlsx)

The XLSX file was created with the following code: 7324387b12dd2e47bd0c33fc577512b92febcfe6

Reply:
Thanks for spotting it. It will be investigated in https://github.com/handsontable/hyperformula/issues/1055